compiler/next/include/chpl/resolution/resolution-types.h

```diff
@@ -504,79 +571,123 @@ class MostSpecificCandidates {
                        PoiInfo poiInfo)
-    : mostSpecific(std::move(mostSpecific)),
-      exprType(std::move(exprType)),
-      poiInfo(std::move(poiInfo)) {
+    : mostSpecific_(std::move(mostSpecific)),
+      exprType_(std::move(exprType)),
+      poiInfo_(std::move(poiInfo)) {
   }
 
+  /** get the most specific candidates for return-intent overloading */
+  const MostSpecificCandidates &mostSpecific() const { return mostSpecific_; }
+
+  /** type of the call expression */
+  const types::QualifiedType &exprType() const { return exprType_; }
+
+  /** point-of-instantiation scopes used when resolving signature or body */
+  const PoiInfo &poiInfo() const { return poiInfo_; }
+
   bool operator==(const CallResolutionResult& other) const {
-    return mostSpecific == other.mostSpecific &&
-           exprType == other.exprType &&
-           PoiInfo::updateEquals(poiInfo, other.poiInfo);
+    return mostSpecific_ == other.mostSpecific_ &&
+           exprType_ == other.exprType_ &&
+           PoiInfo::updateEquals(poiInfo_, other.poiInfo_);
   }
   bool operator!=(const CallResolutionResult& other) const {
     return !(*this == other);
   }
   void swap(CallResolutionResult& other) {
-    mostSpecific.swap(other.mostSpecific);
-    exprType.swap(other.exprType);
-    poiInfo.swap(other.poiInfo);
+    mostSpecific_.swap(other.mostSpecific_);
+    exprType_.swap(other.exprType_);
+    poiInfo_.swap(other.poiInfo_);
   }
 };
 
 /**
   This type represents a resolved expression.
 */
-struct ResolvedExpression {
+class ResolvedExpression {
+ private:
   // What is its type and param value?
-  types::QualifiedType type;
+  types::QualifiedType type_;
   // For simple (non-function Identifier) cases,
   // the ID of a NamedDecl it refers to
-  ID toId;
+  ID toId_;
 
   // For a function call, what is the most specific candidate,
   // or when using return intent overloading, what are the most specific
   // candidates?
   // The choice between these needs to happen
   // later than the main function resolution.
-  MostSpecificCandidates mostSpecific;
+  MostSpecificCandidates mostSpecific_;
   // What point of instantiation scope should be used when
   // resolving functions in mostSpecific?
-  const PoiScope* poiScope = nullptr;
+  const PoiScope *poiScope_ = nullptr;
 
+ public:
   ResolvedExpression() { }
 
+  /** get the type and param value */
+  const types::QualifiedType &type() const { return type_; }
+
+  /** for simple (non-function Identifier) cases, the ID of a NamedDecl it
+   * refers to */
+  ID toId() const { return toId_; }
+
+  /** For a function call, what is the most specific candidate, or when using
+   * return intent overloading, what are the most specific candidates? The
+   * choice between these needs to happen later than the main function
+   * resolution.
+   */
+  const MostSpecificCandidates &mostSpecific() const { return mostSpecific_; }
+
+  const PoiScope *poiScope() const { return poiScope_; }
```

Comment:
FWIW, `Foo *getTheFoo` is more correct in terms of what the C designers meant `*` to mean in a type context. However, while I can explain it, it has never seemed intuitive to me and I prefer to think about `*` doing a different thing in a type and value context and so I like to write `Foo*`. But anyway I don't have a strong opinion about such style in our current work.

Reply:
I don't really mind either way, this is just what clang-format does by default. The relevant config options are `PointerAlignment` and `ReferenceAlignment` with `Left` or `Right` (or even `Middle`!)

Reply:
I understand the rationale behind `Foo *getTheFoo` and I've seen and used both. I also don't have an opinion about which one is better or worse, I just raised the nit because I feel like most of our code uses `Foo&/* getTheFoo` and my eyes stumble when I read the other form. So really just a call for consistency.

Reply:
Glad to see you're already using clang-format. I should probably set some scripts up to start doing the same.

Reply:
fixed formatting